src/data/staticData/extensionData.ts

```diff
@@ -477,8 +497,46 @@ const extensions: { [key: string]: ExtensionData } = {
       ColonyRole.Architecture,
       ColonyRole.Funding,
     ],
-    enabledExtensionBody: WhitelistExtensionBody,
-    initializationParams: [],
+    enabledExtensionBody: (extensionInitParams) =>
+      WhitelistExtensionBody({ extensionInitParams }),
+    initializationParams: [
+      {
+        paramName: 'policy',
+        validation: yup.string().required(),
+        defaultValue: '',
+        title: MSG.whitelistColonyPolicySelectorTitle,
+        type: ExtensionParamType.ColonyPolicySelector,
+        options: [
+          {
+            value: 'AGREEMENT_ONLY',
+            label: MSG.whitelistColonyPolicySelectorAgreementOnly,
+            name: 'policy',
+            appearance: {
+              theme: 'traditional',
+            },
+            checked: false,
+          },
+          {
+            value: 'KYC_ONLY',
+            label: MSG.whitelistColonyPolicySelectorKYCOnly,
+            name: 'policy',
+            appearance: {
+              theme: 'traditional',
+            },
+            checked: false,
+          },
+          {
+            value: 'KYC_AND_AGREEMENT',
+            label: MSG.whitelistColonyPolicySelectorAgreementAndKYC,
+            name: 'policy',
+            appearance: {
+              theme: 'traditional',
+            },
+            checked: false,
+          },
+        ],
+      },
```

Comment:
So this is more complicated that you might initially think.

For one, the first argument of the whitelist, must always be a `bool`, so the value must always be `true/false`

Second, the logic to determine what type of policy we have, is a bit convoluted from the network side:

```
useApproval = true, no agreement hash -> KYC only
useApproval = false, agreement hash -> Agreement only
useapproval = true, agreement hash -> KYC and agreement
```

So that means that your values, need to look something like this:
```suggestion
      {
        paramName: 'policy',
        validation: yup.string().required(),
        defaultValue: '',
        title: MSG.whitelistColonyPolicySelectorTitle,
        type: ExtensionParamType.ColonyPolicySelector,
        options: [
          {
            value: true,
            label: MSG.whitelistColonyPolicySelectorAgreementOnly,
            name: 'policy',
            appearance: {
              theme: 'traditional',
            },
            checked: false,
          },
          {
            value: false,
            label: MSG.whitelistColonyPolicySelectorKYCOnly,
            name: 'policy',
            appearance: {
              theme: 'traditional',
            },
            checked: false,
          },
          {
            value: true,
            label: MSG.whitelistColonyPolicySelectorAgreementAndKYC,
            name: 'policy',
            appearance: {
              theme: 'traditional',
            },
            checked: false,
          },
        ],
      },
```

Reply:
In that case, I'll use numbers as values: `0` for KYC only, `1` for agreement only, and `2` for agreement and KYC, because if we use `true/false` then the UI won't be able to tell the difference between selecting agreement only and agreement and KYC.

Then when the saga is connected, we just do something like `!!policy` to transform the numbers into boolean before sending them.